src/iosrv/init.c

```diff
@@ -202,9 +208,12 @@ dss_tgt_nr_get(int ncores, int nr)
 	D_ASSERT(ncores >= 1);
 	/* Each system XS uses one core, and each main XS with
 	 * dss_tgt_offload_xs_nr offload XS. Calculate the nr_default
-	 * as the number of main XS based on number of cores.
+	 * as the number of main XS based on number of cores available
+	 * on this NUMA node.  Reduce the number of cores available
+	 * by the starting core offset.
 	 */
-	nr_default = (ncores - dss_sys_xs_nr) / DSS_XS_NR_PER_TGT;
+	nr_default = (((ncores - dss_sys_xs_nr) - dss_core_offset) /
```

Comment:
the dss_core_offset and firstcore related setting (from both daos_io_server and daos_server)  seems should be deprecated/removed  after adding this numa nodes bind setting , to avoid confusing.

Reply:
I agree that we want to remove it, but I think it should be phased out rather than removed right now.  We have other components that need to catch up in CI to take advantage of the NUMA node architecture.  By offering legacy mode and the new mode, it allows to us take advantage of the NUMA node core binding algorithm for systems that support it.

Reply:
I am OK to leave to future to refine it, but it is a kind of confuse to with both first_core and numa_node binding.